test/select.test.js

```diff
@@ -0,0 +1,128 @@
+                    },
+                ],
+                offset: 'offsetABC',
+            });
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .select()
+            .eachPage(function page(records) {
+                records.forEach(function(record) {
+                    expect(record.id).toBe('recordA');
+                    expect(record.get('Name')).toBe('Rebecca');
+                });
+                done();
+            });
+    });
+
+    it('selects records with valid params', function(done) {
+        testExpressApp.set('handler override', function(req, res) {
+            expect(req.method).toBe('GET');
+            expect(req.url).toBe(
+                '/v0/app123/Table?maxRecords=50&sort%5B0%5D%5Bfield%5D=Name&sort%5B0%5D%5Bdirection%5D=desc'
+            );
+            res.json({
+                records: [
+                    {
+                        id: 'recordA',
+                        fields: {Name: 'Rebecca'},
+                        createdTime: '2020-04-20T16:20:00.000Z',
+                    },
+                ],
+                offset: 'offsetABC',
+            });
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .select({maxRecords: 50, sort: [{field: 'Name', direction: 'desc'}]})
+            .eachPage(function page(records) {
+                records.forEach(function(record) {
+                    expect(record.id).toBe('recordA');
+                    expect(record.get('Name')).toBe('Rebecca');
+                });
+                done();
+            });
+    });
+
+    it('selects records filters out invalid parameters and extra arguments without erroring', function(done) {
+        testExpressApp.set('handler override', function(req, res) {
+            expect(req.method).toBe('GET');
+            expect(req.url).toBe('/v0/app123/Table?');
+            res.json({
+                records: [
+                    {
+                        id: 'recordA',
+                        fields: {Name: 'Rebecca'},
+                        createdTime: '2020-04-20T16:20:00.000Z',
+                    },
+                ],
+                offset: 'offsetABC',
+            });
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .select({ignoredParam: 'ignore me'}, {anotherIgnored: 'param'})
+            .eachPage(function page(records) {
+                records.forEach(function(record) {
+                    expect(record.id).toBe('recordA');
+                    expect(record.get('Name')).toBe('Rebecca');
+                });
+                done();
+            });
+    });
+
+    it('selects records errors on invalid parameters', function() {
+        return expect(() => {
+            airtable
+                .base('app123')
+                .table('Table')
+                .select({maxRecords: 'should not be a string'});
```

Comment:
This will pass if the calls to `base` or `table` throw an error. Unlikely, sure, but still worth taking those parts of the chain outside of the assertion function.

Reply:
I added matching of the specific error messages. I think this doesn't exactly address you point. But it does make it more clear that the errors I'm seen being throw are the ones I'm trying to throw. Is this a good enough way to address the issue?

Reply:
Yeah, that works for me